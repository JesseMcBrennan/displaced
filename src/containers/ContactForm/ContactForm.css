.pdf {
  display: flex;
  justify-content: center;
}

iframe {
  margin-top: 30px;
  height: 600px;
  width: 600px;
}