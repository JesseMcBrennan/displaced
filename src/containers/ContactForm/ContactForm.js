import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setMessage } from '../../actions';
import './ContactForm.css';

export class ContactForm extends Component {
  constructor() {
    super();
    this.state={
      name: '',
      body: '',
      signature: '',
      error: false
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]:value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { history } = this.props;
    this.props.setMessage(this.state);
    this.setState({
      name: '',
      body: '',
      signature: ''
    });
    if (this.state.body.length) {
      history.push('/ContactPdf');
    } else {
      alert('Please enter a message');
      this.setState({
        error: true
      });
    }
  }

  render () {
    return (
      <form className="ContactForm" onSubmit={this.handleSubmit}>
        <input  
          name='name'
          value={this.state.name}
          onChange={this.handleChange}
          placeholder='Enter your Name'
          type='text'
        />              
        <textarea 
          rows="4"
          cols="50" 
          name='body'
          value={this.state.body}
          onChange={this.handleChange}
          placeholder='Enter your Message'
          className='inputBody'
        />          
        <input 
          name='signature'
          value={this.state.signature}
          onChange={this.handleChange}
          placeholder='Enter your Signature'
          type='text'
        /> 
        <button>Submit your Message</button>    
      </form>
    );
  }
}

export const mapStateToProps = state => ({
  member: state.searchResults
});

export const mapDispatchToProps = dispatch => ({
  setMessage: createMessage => dispatch(setMessage(createMessage))
});

export default connect(mapStateToProps, mapDispatchToProps)(ContactForm);

ContactForm.propTypes = {
  setMessage: PropTypes.func,
  history: PropTypes.object,
  member: PropTypes.object
};

