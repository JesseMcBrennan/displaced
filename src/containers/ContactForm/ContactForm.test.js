import React from 'react';
import { ContactForm, mapStateToProps, mapDispatchToProps } from './ContactForm';
import { shallow } from 'enzyme';

describe('ContactForm', () => {
  let wrapper;
  let mockHistory;
  let event;
  let mockSetMessage;

  beforeEach(() => {
    mockHistory = { push: jest.fn()};
    event = {
      preventDefault: jest.fn(),
      target: {
        value: 'hello',
        name: 'name'
      }
    };
    mockSetMessage = jest.fn();
    wrapper = shallow(
      <ContactForm history={mockHistory} setMessage={mockSetMessage}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with ', () => {

      const mockState = {
        member:{}
      };

      const expected = {};

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch to props', () => {
      const mockDispatch = jest.fn();
      
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.setMessage();
      expect(mockDispatch).toHaveBeenCalledTimes(1);
    });
  });
  describe('handleSubmit', () => {
    it('should call setMessage', () => {
      
      wrapper.state().name = 'Joel';
      wrapper.state().body = 'blahblahblah';
      wrapper.state().signature = 'Joel';

      wrapper.instance().handleSubmit(event);
      expect(mockSetMessage).toHaveBeenCalled();
    });
  });
  describe('handleChange', () => {
    it('should change the state of the container', () => {
      wrapper
        .find('input')
        .first()
        .simulate('change', event);
      expect(wrapper.state('name')).toEqual('hello');
    });
  });
});