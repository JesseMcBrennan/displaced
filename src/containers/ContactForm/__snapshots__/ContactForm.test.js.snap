// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactForm
    history={
        Object {
            "push": [Function],
          }
    }
    setMessage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="name"
          onChange={[Function]}
          placeholder="Enter your Name"
          type="text"
          value=""
/>,
        <textarea
          className="inputBody"
          cols="50"
          name="body"
          onChange={[Function]}
          placeholder="Enter your Message"
          rows="4"
          value=""
/>,
        <input
          name="signature"
          onChange={[Function]}
          placeholder="Enter your Signature"
          type="text"
          value=""
/>,
        <button>
          Submit your Message
</button>,
      ],
      "className": "ContactForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "placeholder": "Enter your Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "inputBody",
          "cols": "50",
          "name": "body",
          "onChange": [Function],
          "placeholder": "Enter your Message",
          "rows": "4",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "signature",
          "onChange": [Function],
          "placeholder": "Enter your Signature",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit your Message",
        },
        "ref": null,
        "rendered": "Submit your Message",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="Enter your Name"
            type="text"
            value=""
/>,
          <textarea
            className="inputBody"
            cols="50"
            name="body"
            onChange={[Function]}
            placeholder="Enter your Message"
            rows="4"
            value=""
/>,
          <input
            name="signature"
            onChange={[Function]}
            placeholder="Enter your Signature"
            type="text"
            value=""
/>,
          <button>
            Submit your Message
</button>,
        ],
        "className": "ContactForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter your Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "inputBody",
            "cols": "50",
            "name": "body",
            "onChange": [Function],
            "placeholder": "Enter your Message",
            "rows": "4",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "signature",
            "onChange": [Function],
            "placeholder": "Enter your Signature",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit your Message",
          },
          "ref": null,
          "rendered": "Submit your Message",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
